storage/innobase/handler/handler0alter.cc

```diff
@@ -665,6 +677,83 @@ check_v_col_in_order(
 	return(true);
 }
 
+/** Get the list of keys re-created because of extending varchar columns.
+@param[in]	ha_alter_info	Data used during inplace alter
+@param[in]	table	MariaDB table
+@return	a vector of old and new key pairs */
+static rename_keys_vector_t
+innobase_get_change_field_keys(Alter_inplace_info* ha_alter_info,
+			       const TABLE* table)
+{
+	rename_keys_vector_t result;
+
+	result.reserve(std::min(ha_alter_info->index_drop_count,
+				ha_alter_info->index_add_count));
+	for (ulint i = 0; i < ha_alter_info->index_drop_count; i++) {
+		const KEY* drop_key = ha_alter_info->index_drop_buffer[i];
+
+		for (ulint j = 0; j < ha_alter_info->index_add_count; j++) {
+			const KEY* add_key = &ha_alter_info->key_info_buffer[
+					ha_alter_info->index_add_buffer[j]];
+
+			if (key_has_changed(drop_key, add_key,
+					    ha_alter_info->alter_info, table)) {
+				continue;
+			}
+
+			// This is ensured by server code.
+			DBUG_ASSERT(lex_string_cmp(system_charset_info,
+						   &add_key->name,
+						   &drop_key->name)
+				    != 0);
+
+			rename_key_pair_t p = {drop_key, add_key};
+			result.push_back(p);
+			break;
+		}
+	}
+
+	return result;
+}
```

Comment:
I would prefer to do this in the SQL layer, and have the `ALTER_RENAME_INDEX` flag.
In that way, the index renaming could be made instantaneously for all storage engines that do not store any index names; I’d expect MyISAM, Aria, Memory and possibly MyRocks to fall in this category.

Reply:
I've changed code to use `ALTER_RENAME_INDEX`. Seems to work. Obviously code became much simpler.